go-controller/pkg/ovn/policy.go

```diff
@@ -151,7 +148,11 @@ func getACLMatch(portGroupName, match string, policyType knet.PolicyType) string
 		aclMatch = "inport == @" + portGroupName
 	}
 
-	if match != "" {
+	if match == "hairpin" {
+		aclMatch += " && (ip4.src == 169.254.169.3 || ip6.src ==  fd69::3)"
```

Comment:
Yeah.. these IPs need to be constants somewhere :). Otherwise they're pure magic.

Reply:
also, add a docblock to this function that the match of "hairpin" is special constant.

Reply:
Good Call will do!